### PR TITLE
Derrick catherine/refund cancellation fix updates

### PR DESCRIPTION
**Pull request by xiec1377**
## Notion ticket link
(no notion ticket) 

1. visual differences:
(a) include “Camp Cancellation” in heading
(b) textStyle for subheading is different
(c) no total refund at the bottom

2. functional differences
(a) doesn’t fail gracefully on bad refund code (FE tries to map refunds which doesn’t exist and crashes, we probably want this to redirect to main page with a toast about the failure)
(b) EDLP should be hidden if it’s price is $0
(c) having a loading spinner on this would be a “nice to have”, search the code base for “spinner” for examples

3. code style:
(a) we’ll want to use existing textStyles rather than hardcoding new font sizes (see [my draft PR](https://github.com/uwblueprint/focus-on-nature/blob/bd781a2c9a18bcf4fcedcca31f88ec585c99f2d7/frontend/src/components/pages/CamperRefundConfirmation/RefundConfirmationCard.tsx#L60) for example)
(b) lots of lint warnings about types and unused variables, we want to clean those up before they hit the main branch

## Implementation description
* For visual differences, added textStyle and button sizing for dynamic page sizing
* For functional differences, 
   * `loading` state first set to true which displays spinner; in `useEffect()` function, if valid API call, set `validCode` to true; else set `validCode` to false, display toast error and return to home page; finally set `loading` to false
   * if `getEDLPTotalCharge()` = $0, then hide EDLP section

## Steps to test
1. Go to [link](https://focus-on-nature.web.app/refund/73cf6242a9dfbdd2) to test for the above requirements

## What should reviewers focus on?
* Test functional bugs 
* Check if visually looks the same to the Figma for all `sm`, `md`, `lg` sizing

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/CamperRefundCancellation/index.tsx [hunk 1]
```diff
@@ -25,27 +28,58 @@ const CamperRefundCancellation = (): React.ReactElement => {
...
-      const getResponse = await CamperAPIClient.getRefundInfo(code);
-      if (getResponse) {
+      try {
+        const getResponse = await CamperAPIClient.getRefundInfo(code);
+        setValidCode(true);
         setRefunds(getResponse);
         setCampName(getResponse[0].campName);
-      } else {
+      } catch {
+        setValidCode(false);
```
Comment (williamtran10) [inline frontend/src/components/pages/CamperRefundCancellation/index.tsx:44]: Might not be necessary if it is initialized at false and never goes from true to false

File: frontend/src/components/pages/CamperRefundCancellation/index.tsx [hunk 2]
```diff
@@ -25,27 +28,58 @@ const CamperRefundCancellation = (): React.ReactElement => {
...
+
+  if (loading) {
+    return (
+      <Center p="30px">
+        <Spinner size="lg" />
+      </Center>
+    );
+  }
+
+  if (validCode === false) {
```
Comment (williamtran10) [inline frontend/src/components/pages/CamperRefundCancellation/index.tsx:79]: Why not `if (!validCode)`

File: frontend/src/components/pages/CamperRefundCancellation/index.tsx [hunk 3]
```diff
@@ -25,27 +28,58 @@ const CamperRefundCancellation = (): React.ReactElement => {
...
       }
     };
     getRefundInfoById(refundCode);
-  }, [toast]);
+  }, []);
+
+  const getTotalRefund = () => {
+    let totalRefund = 0;
+    refunds.forEach((refund, i) => {
+      refund.instances.forEach((instance, j) => {
```
Comment (williamtran10) [inline frontend/src/components/pages/CamperRefundCancellation/index.tsx]: Are `i` and `j` used at all?

File: frontend/src/components/pages/CamperRefundCancellation/index.tsx [hunk 4]
```diff
@@ -61,13 +95,36 @@ const CamperRefundCancellation = (): React.ReactElement => {
         />
         <Box pr={{ lg: "30%", sm: "12%", md: "12%" }} pl="12%" pt="5%">
           <Box pb="40px">
-            <Text as="b" mt="70px" fontSize="40px">
-              {campName}
+            <Text
+              as="b"
```
Comment (williamtran10) [inline frontend/src/components/pages/CamperRefundCancellation/index.tsx:98]: Is this still necessary with `textStyle`? (other instances of same thing in this file too)

File: frontend/src/components/pages/CamperRefundCancellation/CamperRefundFooter.tsx [hunk 5]
```diff
@@ -1,30 +1,35 @@
...
         height="48px"
         variant="primary"
-        background={true ? "primary.green.100" : "primary.green_disabled.100"}
-        mr="50px"
+        background="primary.green.100"
+        textStyle="buttonSemiBold"
+        pt="12px"
         pr="25px"
+        pb="12px"
         pl="25px"
```
Comment (williamtran10) [inline frontend/src/components/pages/CamperRefundCancellation/CamperRefundFooter.tsx:30]: Nit: use `px` and `py` if setting horizontal or vertical padding to same thing

**Review by williamtran10 — CHANGES_REQUESTED** [review]
Overall, good job addressing the big list of changes. Preferably in the future if you make linting changes to files that are out of the scope of the ticket, include them all in a separate PR (thanks for fixing them though).

**Comment by williamtran10** [comment]
Also still a couple small lint changes to make

File: frontend/src/components/pages/CamperRefundCancellation/index.tsx (hunk 4 above)
Comment (derrick-yyang) [reply frontend/src/components/pages/CamperRefundCancellation/index.tsx]: Good catch, I removed it

**Comment by derrick-yyang** [comment]
> Also still a couple small lint changes to make

I think the last lint warnings should be left. One of them is related to `key` which is required from using the `.maps` function.
The other one is unrelated to our changes, not sure if we actually want it or not

**Review by williamtran10 — CHANGES_REQUESTED** [review]
Only a couple of small changes left. Also in the future, it's a good idea to let the reviewer mark conversations as resolved so that it's easier for them to check any changes that were requested.

File: frontend/src/components/pages/CamperRefundCancellation/CamperRefundInfoCard.tsx [hunk 6]
```diff
@@ -6,14 +6,11 @@ import {
...
   Flex,
-  Spacer,
-  extendTheme,
-  ChakraProvider,
 } from "@chakra-ui/react";
 import { CamperRefundDTO, RefundDTO } from "../../../types/CamperTypes";
 
 type CamperRefundInfoCardProps = {
-  camperRefund: any;
+  camperRefund: RefundDTO;
```
[the remark this answers is not in the document]
Comment (williamtran10) [reply frontend/src/components/pages/CamperRefundCancellation/CamperRefundInfoCard.tsx:13]: You can see examples if you search for `key=` in the codebase

**Comment by williamtran10** [comment]
Also double check the most recent commit [637f5c5](https://github.com/uwblueprint/focus-on-nature/pull/289/commits/637f5c553868409f6ab6b85ebde7d6eb0264bb35), it might have reverted some of your fixes

**Review by williamtran10 — APPROVED** [review]
Good work and good job getting through this slog of a ticket. Feel free to merge

**Pull request merged**